src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -202,6 +304,18 @@ static TypeSyntax TryGetSingleTypeArgument(TypeSyntax type)
                 }
                 return null;
             }
+
+            static TypeSyntax ParameterTypeSyntax(IParameterSymbol parameterSymbol)
+            {
+                if (parameterSymbol is object &&
+                    parameterSymbol.DeclaringSyntaxReferences[0].GetSyntax() is ParameterSyntax parameterSyntax &&
+                    parameterSymbol.ContainingSymbol is IMethodSymbol method &&
```

Comment:
oh boy.  can this ever not be true? (the .ContainingSymbol is IMethodSymbol bit)?

Reply:
I don't think it can be, but it's convenient and safe to write this way. If you think it's confusing, I'll move to be a local declaration.